### ISSUE TITLE
Unclosed tags introduced in version 1.7.2

### ISSUE THREAD

**Issue by AronVelling** [zzzprojects/html-agility-pack#217]
Test code:

```
string input = ...
HtmlDocument htmlDoc = new HtmlDocument(){OptionOutputAsXml = true};
htmlDoc.LoadHtml(html);
string output = htmlDoc.DocumentNode.InnerHtml

input <li>a<li>b<li>c
output (1.7.1) <li>a</li><li>b</li><li>c</li>
output (>1.7.1) <li>a<li>b<li>c</li>

input <ul><li>a<li>b<li>c</ul>
output (1.7.1) <ul><li>a</li><li>b</li><li>c</li></ul> 
output (>1.7.1) <ul><li>a<li>b<li>c</li></ul> 
```

I think the pre 1.7.2 output was correct

**Comment by JonathanMagnan** [comment]
Hello @AronVelling ,

Is this cause an issue? Not closing `li` is valid if the next open tag is another `li`.

That's why you see only the latest `li` with a closing tag because there is a `ul`.

Best Regards,

Jonathan

---

**Performance Libraries**
`context.BulkInsert(list, options => options.BatchSize = 1000);  `
[Entity Framework Extensions](http://entityframework-extensions.net/) • [Bulk Operations](http://bulk-operations.net/) •  [Dapper Plus](http://dapper-plus.net/) •  [LinqToSql Plus](http://linqtosql-plus.net/)

**Runtime Evaluation**
`Eval.Execute("x + y", new {x = 1, y = 2}); // return 3`
[C# Eval Function](http://eval-expression.net/) • [SQL Eval Function](http://eval-sql.net/)

**Comment by AronVelling** [comment]
I use the OptionOutputAsXml = true option
And the output is not valid xml. 
So at least when OptionOutputAsXml = true the output should be as it was pre 1.7.2

**Comment by AronVelling** [comment]
Don't you agree that this is incorrect when OptionOutputAsXml = true?

**Comment by JonathanMagnan** [comment]
Hello @AronVelling ,

Sorry for the delay, we had a new project released last week so it took us more time than expected and we didn't have much time for free library.

I agree 100% with you.

It's something we will try to do and release today.

Best Regards,

Jonathan

**Comment by AronVelling** [comment]
Nice. 
Thanks

**Comment by JonathanMagnan** [comment]
Hello @AronVelling ,

The v1.8.5 has been released.

Be careful, the option `OptionOutputAsXml` as been created when saving in a file such as

```csharp
using (StringWriter writer = new StringWriter())
{
    htmlDoc.Save(writer);

    var s = writer.ToString();
}
```

However, I also included a fix that when the option is enabled and a tag is flagged as `IsImpliciedEnd` to change it for `IsExplicitEnd`, so it should work as well with your code.

Let me know if everything works as expected.

Best Regards,

Jonathan
